include/seqan3/search/dream_index/detail/bitvector.hpp

```diff
@@ -0,0 +1,1194 @@
+        return lhs.data() != rhs.data();
+    }
+
+    /*!\brief Test for less than.
+     * \param lhs First instance.
+     * \param rhs Instance to compare to.
+     * \returns `true` if this is smaller than rhs, `false` otherwise.
+     */
+    friend bool operator<(bitvector_uncompressed const & lhs, bitvector_uncompressed const & rhs) noexcept
+    {
+        return lhs.data() < rhs.data();
+    }
+
+    /*!\brief Test for greater than.
+     * \param lhs First instance.
+     * \param rhs Instance to compare to.
+     * \returns `true` if this is greater than rhs, `false` otherwise.
+     */
+    friend bool operator>(bitvector_uncompressed const & lhs, bitvector_uncompressed const & rhs) noexcept
+    {
+        return lhs.data() > rhs.data();
+    }
+
+    /*!\brief Test for less than or equal.
+     * \param lhs First instance.
+     * \param rhs Instance to compare to.
+     * \returns `true` if this is less than or equal to rhs, `false` otherwise.
+     */
+    friend bool operator<=(bitvector_uncompressed const & lhs, bitvector_uncompressed const & rhs) noexcept
+    {
+        return lhs.data() <= rhs.data();
+    }
+
+    /*!\brief Test for greater than or equal.
+     * \param lhs First instance.
+     * \param rhs Instance to compare to.
+     * \returns `true` if this is greater than or equal to rhs, `false` otherwise.
+     */
+    friend bool operator>=(bitvector_uncompressed const & lhs, bitvector_uncompressed const & rhs) noexcept
+    {
+        return lhs.data() >= rhs.data();
+    }
+    //!\}
+
+    /*!\brief Swap contents with another instance.
+     * \param rhs The other instance to swap with.
+     *
+     * \par Complexity
+     *
+     * Constant.
+     *
+     * \par Exceptions
+     *
+     * No-throw guarantee.
+     */
+    void swap(bitvector_uncompressed & rhs) noexcept
+    {
+        std::swap(data_value, rhs.data_value);
+    }
+
+    //!\overload
+    void swap(bitvector_uncompressed && rhs) noexcept
+    {
+        std::swap(data_value, rhs.data_value);
+    }
+
+    /*!\brief Swap contents with another instance.
+     * \param lhs  The first instance.
+     * \param rhs The other instance to swap with.
+     *
+     * \par Complexity
+     *
+     * Constant.
+     *
+     * \par Exceptions
+     *
+     * No-throw guarantee.
+     */
+    friend void swap(bitvector_uncompressed & lhs, bitvector_uncompressed & rhs) noexcept
+    {
+        lhs.swap(rhs);
+    }
+
+    //!\overload
+    friend void swap(bitvector_uncompressed && lhs, bitvector_uncompressed && rhs) noexcept
+    {
+        lhs.swap(rhs);
+    }
+
+    /*!\brief Assign a value to a position in the bitvector.
+     * \param index The position to assign to.
+     * \param value The value to assign.
+     *
+     * \attention This method will set as many bits as there are in `size_type`, i.e. usually 64 bits.
+     *
+     * \par Complexity
+     *
+     * Constant.
+     */
+    void set_int(size_type const index, size_type const value)
```

Comment:
three questions:  
1. what value are you assigning? it is a bitvector so its value_type are bits?
2. is the set_int of data_value private or protected? 
3. what was your motivation of wrapping the sdsl vector instead of inheriting from it?

Reply:
1. we set 64 bits at once
2. public
3. Mhm, the SDSL bitvector models ReservableContainer, but I wanted to add some asserts. But I could probably just inherit and be done with it?

Reply:
well, then this is a lot of code that just delegates to the underlying data member which would not be neccessary if you inherit (e.g. size(), max_size()), so if there is no reason against it I would say inheritance is better in this case and you overload only those functions where asserts are needed/useful. What do you think?